### PR TITLE
Feature systems

### PR DESCRIPTION
**Pull request by DnMllr**
I am submitting a cleanup of core that has a few notable features.

1. Transforms are calculated far more efficiently.
2. Everything is based around paths representing a precise location in the scene graph.
3. Node's internal logic has been greatly simplified.
4. Unit testing patterns have been notably improved.

This will be ready to merge once it passes travis.

Thanks for everyone's patience.

### CODE REVIEW

File: core/SizeSystem.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+ *
+ * @method
+ * @private
+ *
+ * @param {Node} node Node to potentially call onDifferentialSize on
+ * @param {Array} components a list of the nodes' components
+ * @param {Size} the size component
+ */
+function differentialSizeChanged (node, components, size) {
+    var differentialSize = size.getDifferentialSize();
```
Comment (bart-jansen) [inline core/SizeSystem.js]: @DnMllr `size` doesn't seem to have a `getDifferentialSize()` method. Using`node` and changing this to `var differentialSize = node.getDifferentialSize();` seems to work well though.

**Comment by bart-jansen** [comment]
Really big :+1: I did some performance testing using 900(!) rotating DIVs. Using the current engine I'd be able to get **2-3FPS** whereas this rewrite performs at **55-60(!)FPS**

Great job @DnMllr !

File: core/SizeSystem.js (hunk 1 above)
Comment (alexanderGugel) [inline core/SizeSystem.js]: @DnMllr https://github.com/DnMllr/engine/pull/10

File: package.json [hunk 2]
```diff
@@ -46,7 +51,8 @@
   },
   "browserify": {
     "transform": [
-      "glslify"
+      "glslify",
+      "cssify"
```
Comment (michaelobriena) [inline package.json]: This is killing the travis build.  We do not have a dep on cssify anymore.

Comment (TheAlphaNerd) [inline package.json]: Has this been rebased against develop?

There are regressions in this package json

For example the removal of the tape script

Comment (alexanderGugel) [inline package.json]: https://github.com/DnMllr/engine/pull/6

**Comment by michaelobriena** [comment]
Landed in 0.6

**Pull request closed without merging**